### DISCUSSION TITLE
Trouble setting up self-registration following docs / adding a default 'public-facing' site setup option?

### DISCUSSION THREAD

**Discussion by kylebakerio** [requarks/wiki#3024] [Help / Questions]
I'm attempting to enable the ability for users to self-signup following these instructions:

https://docs.requarks.io/auth#self-registration

However, when I then try to sign up, I get a red popup that says, 

`The mail configuration is incomplete or invalid`

Which appears to correspond to error 3002 from this page:

https://docs.requarks.io/dev/api

`3002	MailNotConfigured	The mail configuration is incomplete or invalid.`

When I later try to sign up with that email again, I get notified that the user already exists, but when I attempt to sign in, I am notified that I must confirm my account.

It just so happens that I already had created a 'user group' to use for self-registered users, but it might make sense to add that there as well--as well as to have a default setting for all of this to 'just work' as a default public-facing wiki in the initial site setup flow? For now I'll go dig further into setting up mailing stuff.

But it's not ideal that there isn't a mention of that there in the docs.

**Comment by NGPixel** [comment]
You must setup the mail settings under Admin > Mail so that a verification email can be sent during self-registration. There's actually a warning in the logs when starting the Wiki.js process.

**Comment by kylebakerio** [comment]
I dug into it more and realized I'd rather just use oauth, and found the guide for github to be great. there was no guide for gitlab, but I was able to figure it out using the github process as a guide.

Mail server stuff is stuff I touch once in a blue moon and then forget how it works after a few years when I come to it again. But I still think it would be nice if there was a 'set up for self-registration public-facing mode' flow that guided you through things like creating a group that suits them, setting up mail server and/or oauth, etc.

I'm not watching the wikijs server logs closely and wouldn't expect to be unless something went wrong--should I think of it differently?
